python/meterpreter/ext_server_stdapi.py

```diff
@@ -2146,6 +2150,75 @@ def stdapi_net_config_get_routes(request, response):
         response += tlv_pack(TLV_TYPE_NETWORK_ROUTE, route_tlv)
     return ERROR_SUCCESS, response
 
+def windll_route_add_remove(is_add, request, response):
+    class IPAddr(ctypes.Structure):
+        _fields_ = [
+            ("S_addr", ctypes.c_ulong)]
+
+    def inet_addr(ip):
+        return IPAddr(struct.unpack("L", socket.inet_aton(ip))[0])
+
+    MIB_IPROUTE_TYPE_INDIRECT = 4
+    MIB_IPPROTO_NETMGMT = 3
+
+    GetBestInterface = ctypes.windll.Iphlpapi.GetBestInterface
+    GetBestInterface.argtypes = [IPAddr, ctypes.POINTER(ctypes.c_ulong)]
+    GetBestInterface.restype = ctypes.c_ulong
+
+    CreateIpForwardEntry = ctypes.windll.Iphlpapi.CreateIpForwardEntry
+    CreateIpForwardEntry.argtypes = [PMIB_IPFORWARDROW]
+    CreateIpForwardEntry.restype = ctypes.c_ulong
+
+    DeleteIpForwardEntry = ctypes.windll.Iphlpapi.DeleteIpForwardEntry
+    DeleteIpForwardEntry.argtypes = [PMIB_IPFORWARDROW]
+    DeleteIpForwardEntry.restype = ctypes.c_ulong
+
+    GetIpInterfaceEntry = ctypes.windll.Iphlpapi.GetIpInterfaceEntry
+    GetIpInterfaceEntry.argtypes = [ctypes.POINTER(MIB_IPINTERFACE_ROW)]
+    GetIpInterfaceEntry.restype = ctypes.c_ulong
+
+    subnet = packet_get_tlv(request, TLV_TYPE_SUBNET_STRING)['value']
+    netmask = packet_get_tlv(request, TLV_TYPE_NETMASK_STRING)['value']
+    gateway = packet_get_tlv(request, TLV_TYPE_GATEWAY_STRING)['value']
+
+    route = MIB_IPFORWARDROW()
+    route.dwForwardDest = struct.unpack("I", socket.inet_aton(subnet))[0]
+    route.dwForwardMask = struct.unpack("I", socket.inet_aton(netmask))[0]
+    route.dwForwardNextHop = struct.unpack("I", socket.inet_aton(gateway))[0]
```

Comment:
This would be more clear if we always correctly unpacked the integer as big-endian which is what `inet_aton` always returns. I'd put `addr_atoi4` outside this function towards the top as a utility. It'll convert an IP address into a 32-bit integer in network byte order. After that you just need to convert it from network to host byte order if you're passing it to the Windows API with `socket.ntohl` (network-to-host-long).

```
def addr_atoi4(address):
    return struct.unpack('!I',  socket.inet_aton(address))[0]
...
route.dwForwardDest = socket.ntohl(addr_atoi4(address))
...
```

This should ensure it works on systems that are little endian as well as be more clear.